src/main/java/gregtech/api/pipenet/block/BlockPipe.java

```diff
@@ -335,15 +327,22 @@ public boolean onPipeActivated(World world, IBlockState state, BlockPos pos, Ent
 
         EnumActionResult result = coverBehavior.onRightClick(entityPlayer, hand, hit);
         if (result == EnumActionResult.PASS) {
-            if (pipeTile.getFrameMaterial() != null) {
-                BlockFrame blockFrame = MetaBlocks.FRAMES.get(pipeTile.getFrameMaterial());
-                return blockFrame.onBlockActivated(world, pos, state, entityPlayer, hand, hit.sideHit, (float) hit.hitVec.x, (float) hit.hitVec.y, (float) hit.hitVec.z);
+            if (activateFrame(world, state, pos, entityPlayer, hand, hit, pipeTile)) {
+                return true;
```

Comment:
Why not do `return activateFrame()` like all the other cases?

Reply:
because when it fails it should try screwdriver click in the cover